fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/domain/SavingsAccountCharge.java

```diff
@@ -474,7 +475,7 @@ public void update(final BigDecimal amount, final LocalDate dueDate, final Month
                 break;
                 case PERCENT_OF_AMOUNT:
                     this.percentage = newValue;
-                    this.amountPercentageAppliedTo = null;
+                    this.amountPercentageAppliedTo = BigDecimal.ZERO;
```

Comment:
Hoping this is not check for null somewhere in the code base?

Reply:
Can't tell for sure, Just know it's a nullable field and as null, spotbugs throws an error when the value is passed to `update.percentageOf(BigDecimal, BigDecimal)`. So I just replaced that particular instance where I needed it to be something other than null

Reply:
I think we cannot have this zero, since it would affect the calculation, there is a reason why they were using null, Zero means there is some value provided ie zero, did you test it though?

Reply:
Hmm, This is troublesome. I thought this was the most subtle way of going around this. I don't think multiplication by null is possible. This is actually some faulty method but I think I'll see how to make spotsbug ignore this just to stay on the safe side

Reply:
Yes true, might be just initializing it with null and then populating with real values... Could be like that